File: 프로그래머스/lv1/42840. 모의고사/모의고사.java
import java.util.ArrayList;
class Solution {
    public int[] solution(int[] answers) {
        ArrayList<Integer> al=new ArrayList<>();
        int[] arr1={1,2,3,4,5};
        int[] arr2={2,1,2,3,2,4,2,5};
        int[] arr3={3,3,1,1,2,2,4,4,5,5};
        int[] answerCount={0,0,0};
        
        for(int i=0;i<answers.length;i++){
            if(arr1[i%5]==answers[i]) answerCount[0]++;
            if(arr2[i%8]==answers[i]) answerCount[1]++;
            if(arr3[i%10]==answers[i]) answerCount[2]++;
        }
        int max=answerCount[0];
        int size=0;
        for(int i=1;i<answerCount.length;i++){
            if(max<answerCount[i]){
                max=answerCount[i];
            }
        }
        
        
        for(int i=0;i<answerCount.length;i++){
            if(answerCount[i]==max){
                al.add(i+1);
            }
        }
        int[] answer = new int[al.size()];
        for(int i=0;i<al.size();i++){
            answer[i]=al.get(i);
        }
        
        
        
        return answer;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
a = [1, 2, 3, 4, 5]
b = [2, 1, 2, 3, 2, 4, 2, 5]
c = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]



def solution(answers):
    answer = []
    aa = [1 for i, val in enumerate(answers) if a[i % 5] == val]
    bb = [1 for i, val in enumerate(answers) if b[i % 8] == val]
    cc = [1 for i, val in enumerate(answers) if c[i % 10] == val]

    x = max([len(aa),len(bb),len(cc)])

    if x == len(aa) :
        answer.append(1)
    if x == len(bb) :
        answer.append(2)
    if x == len(cc) :
        answer.append(3)

    sorted(answer)

    return answer